mkosi/bootloader.py

```diff
@@ -585,6 +585,9 @@ def find_and_install_shim_binary(
 
                 rel = p.relative_to(context.root)
                 if (context.root / output).is_dir():
+                    # The ESP wants .efi files, not .efi.signed
+                    if rel.suffix == ".signed":
+                        rel = rel.with_suffix("")
```

Comment:
I looked at the callsite once more and it's got one more edge case: For Ubuntu this seems to be `.efi.signed.latest`
```suggestion
                    # The ESP wants .efi files, not .efi.signed or .efi.signed.latest
                    if rel.suffix and rel.suffix != ".efi":
                        left_stem, _ = p.name.split(".", maxsplit=1)
                        rel = rel.with_name(f"{left_stem}.efi")
```